lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -68,4 +99,97 @@ def desired_options
       AWSResource.lookup_options(options, resource: new_resource)
     end
   end
+
+  def update_attachment(volume)
+    status = volume.status
+    #
+    # If we were told to attach the volume to a machine, do so
+    #
+    if desired_instance.is_a?(AWS::EC2::Instance)
+      case status
+      when :in_use
+        attachment = volume.attachments.first
+        # wrong instance
+        if attachment.instance != desired_instance
+          detach(volume, :instance => attachment.instance, :device => attachment.device)
+          attach(volume)
+        # wrong device only
+        elsif attachment.instance == desired_instance && attachment.device != new_resource.device
+          detach(volume, :device => attachment.device)
+          attach(volume)
+        end
+
+      when :available
+        attach(volume)
+      when nil
+        raise "EBS volume #{new_resource.name} does not currently exist!"
```

Comment:
I ran into an issue if `get_driver_and_id` is not defined like on aws_key_pair.  

I did this. Better way?
```ruby
  def to_s
    id = nil
    id = get_driver_and_id[1] if self.respond_to?(:get_driver_and_id)
    "#{declared_type}[#{@name}] (#{ id ? id : 'no AWS object id'})"
  end
```